projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/pc_sampling/ioctl/ioctl_adapter.cpp

```diff
@@ -578,6 +615,9 @@ ioctl_pcs_create(const rocprofiler_agent_t*       agent,
                  uint64_t                         interval,
                  uint32_t*                        ioctl_pcs_id)
 {
+    if(check_firmware_compatibility(agent, method) != ROCPROFILER_STATUS_SUCCESS)
+        return ROCPROFILER_STATUS_INCOMPATIBLE_FIRMWARE;
+
```

Comment:
Just nit-picking. Looks to me that the check_firmware_compatibility can return two different status codes, so I think something like this would be more appropriate

```suggestion
if (auto status = check_firmware_compatibility(agent, method);
    status != ROCPROFILER_STATUS_SUCCESS)
    return status ;
```